### ISSUE TITLE
mz-datepicker not detected change dates

### ISSUE THREAD

**Issue by LucasViniciusPereira** [sherweb/ng2-materialize#285]
I'm using mz-datepicker for dynamic load data, but it does not have correct functionality because I fetch the data from the service and then assign the component's disable options. 

Example:
```html
<mz-datepicker-container class="col s12">
  <input mz-datepicker [options]="options" [formControl]="Date" />
</mz-datepicker-container>
```

```javascript
public dates = [];
public options: Pickadate.DateOptions = { };

constructor() {
  this.options.disable = this.dates;

  Observable.timer(5000).subscribe(() => {
    this.dates = [true, new Date('2018, 02, 07'), new Date('2018, 02, 06'), new Date('2018, 02, 05'), new Date('2018, 02, 04')];

    this.options.disable = this.dates;
  });
}

```

I having disabled dates dynamic, because of business rules.

**Comment by jfcere** [comment]
@LucasViniciusPereirat thanks for reporting the issue, i'll look into it and come back to you.

**Comment by jfcere** [comment]
I submitted a PR to fix this issue, it will be available in the new v1.8.0 release coming soon.

Note that because the `options` input property is an object, you will need to assign a new object for detection changes to triggered correctly and for the changes to be applied to the datepicker.

Here is the way to options will need to be modified by creating a new object using destructor...
```javascript
this.dates = [true, new Date('2018, 02, 07'), new Date('2018, 02, 06'), new Date('2018, 02, 05'), new Date('2018, 02, 04')];

this.options = {
  ...this.options,
  disable: this.dates,
};
```

**Comment by LucasViniciusPereira** [comment]
Thanks @jfcere , I will waiting for new release.
